modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/CommonBeanDefinitionCreator.java

```diff
@@ -0,0 +1,260 @@
+            .add(MESSAGE_FILTER_ELEMENT_IDENTIFIER)
+            .add(MULE_IDENTIFIER)
+            .add(DEFAULT_ES_ELEMENT_IDENTIFIER)
+            .build();
+
+    @Override
+    public boolean handleRequest(CreateBeanDefinitionRequest request)
+    {
+        ComponentModel componentModel = request.getComponentModel();
+        ComponentBuildingDefinition buildingDefinition = request.getComponentBuildingDefinition();
+        componentModel.setType(retrieveComponentType(componentModel, buildingDefinition));
+        BeanDefinitionBuilder beanDefinitionBuilder = createBeanDefinitionBuilder(componentModel, buildingDefinition);
+        processAnnotations(componentModel, beanDefinitionBuilder);
+        processComponentDefinitionModel(request.getParentComponentModel(), componentModel, buildingDefinition, beanDefinitionBuilder);
+        return true;
+    }
+
+    private BeanDefinitionBuilder createBeanDefinitionBuilder(ComponentModel componentModel, ComponentBuildingDefinition buildingDefinition)
+    {
+        BeanDefinitionBuilder beanDefinitionBuilder;
+        if (buildingDefinition.getObjectFactoryType() != null)
+        {
+            beanDefinitionBuilder = createBeanDefinitionBuilderFromObjectFactory(buildingDefinition);
+        }
+        else
+        {
+            beanDefinitionBuilder = genericBeanDefinition(componentModel.getType());
+        }
+        return beanDefinitionBuilder;
+    }
+
+    private void processAnnotationParameters(ComponentModel componentModel, Map<QName, Object> annotations)
+    {
+        componentModel.getParameters().entrySet().stream().filter(entry -> {
+            return entry.getKey().contains(":");
+        }).forEach(annotationKey -> {
+            Node attribute = from(componentModel).getNode().getAttributes().getNamedItem(annotationKey.getKey());
+            if (attribute != null)
+            {
+                annotations.put(new QName(attribute.getNamespaceURI(), attribute.getLocalName()), annotationKey.getValue());
+            }
+        });
+    }
+
+    private void processNestedAnnotations(ComponentModel componentModel, Map<QName, Object> previousAnnotations)
+    {
+        componentModel.getInnerComponents().stream().filter(cdm -> {
+            return cdm.getIdentifier().equals(ANNOTATIONS_ELEMENT_IDENTIFIER);
+        }).findFirst()
+                .ifPresent(annotationsCdm -> {
+                    annotationsCdm.getInnerComponents().forEach(annotationCdm -> {
+                                                                    previousAnnotations.put(new QName(from(annotationCdm).getNamespaceUri(), annotationCdm.getIdentifier().getName()), annotationCdm.getTextContent());
+                                                                }
+                    );
+                });
+    }
+
+    private void processAnnotations(ComponentModel componentModel, BeanDefinitionBuilder beanDefinitionBuilder)
```

Comment:
if I understand correctly this will be executed for any component which supports XML annotations right?